server/routes/health_routes.py

```
@@ -21,33 +20,19 @@
-            sslmode=os.getenv('POSTGRES_SSLMODE', 'prefer') or None,
-            sslrootcert=os.getenv('POSTGRES_SSLROOTCERT') or None,
-        )
-
-        cursor = conn.cursor()
-        # No RLS needed — infrastructure health check
-        cursor.execute("SELECT 1")
-        cursor.close()
-        conn.close()
+        from utils.db.connection_pool import db_pool
+        with db_pool.get_connection() as conn:
+            with conn.cursor() as cursor:
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)